tests/regression/04-mutex/41-pt_rwlock.c

```diff
@@ -19,8 +19,8 @@ int main(void) {
   pthread_create(&id, NULL, t_fun, NULL);
 
   pthread_rwlock_rdlock(&rwlock);
-  printf("%d",data1); //NORACE!
-  data2++;            //RACE
+  printf("%d",data1); // NORACE
+  data2++;            // RACE
```

Comment:
Maybe it would be interesting to add a version where a second thread also does 

```
  pthread_rwlock_rdlock(&rwlock);
  printf("%d",data1); // NORACE
  data2++;            // RACE

```

Because the it would be `RACE!`

Reply:
I will add a few variations on this for submission but I think we aren't planning to add proper read-write lock support which would be able to rule this one out.

There is some slightly different handling of write and read lock locking currently implemented but that is not exactly it. It seems to consider _any_ write in a read lock section potentially unsafe. That is why the upper `NORACE` lines are correctly excluded because the writing happens in the thread with a write lock.